File: leetcode_c_plus_plus/27.cpp
// 27. 移除元素
// https://leetcode-cn.com/problems/remove-element/
using namespace std;
#include <vector>
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int length = nums.size();
        if (length == 0) return 0;
        int left = -1;
        int right = 0;
        while(right < length){
            if(nums[right] != val){
                left++;
                nums[left] = nums[right];
            }
            right++;
        }
        return left+1;
    }
};

File: 剑指offer/python/27.py
# 剑指 Offer 27. 二叉树的镜像
# https://leetcode-cn.com/problems/er-cha-shu-de-jing-xiang-lcof/
from collections import deque
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def mirrorTree(self, root: TreeNode) -> TreeNode:
        if not root:
            return None
        d = deque()
        d.append(root)

        while d:
            node = d.popleft()
            node.right , node.left = node.left,node.right
            if node.right:
                node.append(node.right)
            if node.left:
                node.append(node.left)
        
        return  root